config/config.go

```diff
@@ -191,6 +196,87 @@ func (cfg Config) CallbackURL(path string, mergeParams ...url.Values) string {
 	return base.String()
 }
 
+//  MatchURL will compare two url strings and will return true if they match.
+func MatchURL(baseURL, testURL string) (bool, error) {
+	compareQueryValues := func(baseVal, testVal url.Values) bool {
+		for name := range baseVal {
+			if baseVal.Get(name) == testVal.Get(name) {
+				continue
+			}
+			return false
+		}
+		return true
+	}
+
+	addImplicitPort := func(u *url.URL) {
+		if strings.Contains(u.Host, ":") {
+			return
+		}
+		switch strings.ToLower(u.Scheme) {
+		case "http":
+			u.Host += ":80"
+		case "https":
+			u.Host += ":443"
+		}
+	}
+	base, err := url.Parse(baseURL)
+
+	if err != nil {
+		return false, err
+	}
+	test, err := url.Parse(testURL)
+
+	if err != nil {
+		return false, err
+	}
+
+	addImplicitPort(base)
+	addImplicitPort(test)
+
+	// use full host that includes port by default, i.e example.com:8080
+	baseHost := base.Host
+	testHost := test.Host
+
+	// host/port check
+	if !strings.EqualFold(baseHost, testHost) {
+		return false, nil
+	}
+
+	// scheme check
+	if !strings.EqualFold(base.Scheme, test.Scheme) {
+		return false, nil
+	}
+
+	// path check
+	if len(base.Path) > 1 && !strings.HasPrefix(test.Path, base.Path) {
+		return false, nil
+	}
+
+	// query check
+	if len(test.Query()) > 0 || len(base.Query()) > 0 {
+		if !compareQueryValues(base.Query(), test.Query()) {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+// ValidWebhookURL returns true if the URL is an allowed webhook source.
+func (cfg Config) ValidWebhookURL(whURL string) bool {
+	if len(cfg.Webhook.AllowedURLs) == 0 {
+		return true
+	}
+	for _, u := range cfg.Webhook.AllowedURLs {
+		matched, err := MatchURL(whURL, u)
+		if err != nil {
+			return false
+		}
+		return matched
```

Comment:
only jump out if `matched` is true, otherwise this will reject the first non-match even if allowed via a subsequent entry in `AllowedURLs`